pkgs/applications/audio/soundconverter/default.nix

```diff
@@ -0,0 +1,82 @@
+{ lib, fetchurl
+# Optional due to unfree license.
+, faacSupport ? false
+, glib, python3Packages, gtk3, wrapGAppsHook
+, gsettings-desktop-schemas, intltool, xvfb-run
+, gobject-introspection, gst_all_1, fdk-aac-encoder }:
+
+python3Packages.buildPythonApplication rec {
+  pname = "soundconverter";
+  version = "4.0.3";
+
+  src = fetchurl {
+    url = "https://launchpad.net/soundconverter/trunk/${version}/+download/${pname}-${version}.tar.gz";
+    sha256 = "sha256-hzIG/4LD3705erPYvXb7uoRwF9LtKKIKB3jrhpYMsZ0=";
+  };
+
+  buildInputs = [
+    gtk3
+    fdk-aac-encoder
+    gst_all_1.gst-libav
+    gst_all_1.gst-plugins-base
+    gst_all_1.gst-plugins-good
+    gst_all_1.gst-plugins-ugly
+    (gst_all_1.gst-plugins-bad.override { inherit faacSupport; })
+  ];
+
+  nativeBuildInputs = [
+    intltool
+    wrapGAppsHook
+  ];
+
+  propagatedBuildInputs = [
+    gobject-introspection
+    python3Packages.gst-python
+    python3Packages.distutils_extra
+    python3Packages.setuptools
+  ];
+
+  checkInputs = [
+    xvfb-run
+  ];
+
+  postPatch = ''
+    substituteInPlace  bin/soundconverter --replace \
+      "DATA_PATH = os.path.join(SOURCE_PATH, 'data')" \
+      "DATA_PATH = '$out/share/soundconverter'"
+  '';
+
+  preCheck = let
+    self = { outPath = "$out"; name = "${pname}-${version}"; };
+    xdgPaths = lib.concatMapStringsSep ":" glib.getSchemaDataDirPath;
+  in ''
+    export HOME=$TMPDIR
+    export XDG_DATA_DIRS=$XDG_DATA_DIRS:${xdgPaths [gtk3 gsettings-desktop-schemas self]}
+  '' + lib.optionalString (!faacSupport) ''
+    substituteInPlace tests/testcases/integration.py --replace \
+      "for encoder in ['fdkaacenc', 'faac', 'avenc_aac']:" \
+      "for encoder in ['fdkaacenc', 'avenc_aac']:"
+  '';
+
+  checkPhase = ''
```

Comment:
There are still a handful of tests that fail, I thing a comment should be added to the checkPhase with the reason of why they fail.

Otherwise the software builds and runs. Just tested an audio file convertion.

Reply:
Yeah, the `FilenameToUriTest` appears to fail because [`filename_to_uri()`](https://github.com/kassoulet/soundconverter/blob/f9c1c96225f2027866f1fc7f2dd584b95673d76d/soundconverter/util/fileoperations.py#L129-L157) appears to convert paths incorrectly:

* Ubuntu: `https://example.com/foo` > `http://example.com/foo`
* Nix: `https://example.com/foo` > `file:///build/soundconverter-4.0.3/http:/example.com/foo`

It's clearly broken, but not sure why.

Reply:
Actually that's incorrect, it's [`Gio.file_parse_name()`](https://gitlab.gnome.org/GNOME/glib/blob/2.66.2/gio/gfile.c#L6561-6578) that's causing this, not `filename_to_uri()`.

Reply:
Yeah, there's something wrong with it alright...
```
 > nix-shell -p gobject-introspection
[nix-shell:~]$ python
Python 3.8.12 (default, Aug 30 2021, 16:42:10) 
[GCC 10.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from gi.repository import Gio
>>> Gio.file_parse_name('https://example.com/foo').get_uri()
'file:///home/jakubgs/https:/example.com/foo'
>>> 
```

Reply:
This might be some kind of version mismatch, not sure, don't feel like debugging the `Gio` library.

Reply:
Added a comment.

Reply:
Gio requires GVfs for non-local schemes (`${gvfs}/lib/gio/modules` in `GIO_EXTRA_MODULES`).

Reply:
Interesting, and this is supposed to work at runtime? I tried adding this:
```
export GIO_EXTRA_MODULES=${lib.getLib gvfs}/lib/gio/modules
```
But the test still fails the exact same way.

Reply:
@jtojnar including `gvfs` as a dependency doesn't help:
```
 > nix-shell -p python3Packages.pygobject3 -p gvfs 

[nix-shell:~/work/nixpkgs]$ python
Python 3.9.6 (default, Jun 28 2021, 08:57:49) 
[GCC 10.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from gi.repository import Gio
>>> Gio.file_parse_name('https://example.com/foo').get_uri()
'file:///home/jakubgs/work/nixpkgs/https:/example.com/foo'
```
So not sure how this is supposed to work.

Reply:
Honestly I don't get what's the point of this test:
```python
        for path in ('http://example.com/foo', ):
            uri = filename_to_uri(path)
            self.assertTrue(uri.startswith('http://'))
            self.assertTrue(Gio.file_parse_name(path).get_uri() in uri)
```
Since as far as I can tell SoundConverter is not intended for converting things from online sources.

Reply:
On Ubuntu it works just fine:

![image](https://user-images.githubusercontent.com/2212681/156930904-0df5e374-0c29-440e-b474-07c82d5d60f6.png)

Maybe this is an issue with Nix verison of Gio is built?

Reply:
I've disabled the test as I see no clear way to fix it.